hugegraph-server/hugegraph-core/src/main/java/org/apache/hugegraph/backend/store/AbstractBackendStoreProvider.java

```diff
@@ -67,7 +70,10 @@ protected final void checkOpened() {
 
     @Override
     public void listen(EventListener listener) {
-        this.storeEventHub.listen(EventHub.ANY_EVENT, listener);
+        if (schemaCacheClearListened.compareAndSet(false, true) ||
+            vertexEdgeCacheClearListened.compareAndSet(false, true)) {
```

Comment:
Some question:
- we define 2 automic_param boolean A & B and try to set them to `true`, when to reset it?
- why we need 2 flag to mark it? (in this context)
- since **assignment operations** (`=`) for **primitive types** are atomic by default(64bit), it seems that we only need to use a `volatile` boolean flag to achieve the desired effect?

@Pengzna @VGalaxies

Reply:
1、define 2 automic_param boolean A & B;this operation is to ensure that storeEventListener does not deliver to storeEventHub of storeProvider once in each thread. I understand that there is no need to re register during the entire Hugegraph lifecycle, so there is no reset operation; Will this miss other necessary scenarios?
2、schemaCacheClearListened for schemaCache clear , vertexEdgeCacheClearListened for vertex&edge cache;
3、 I have modified it to use the volatile+synchronized，thanks